### ISSUE TITLE
Potential regression causing resource exhausted after recent commit

### ISSUE THREAD

**Issue by yhavinga** [erfanzar/EasyDeL#99]
I noticed latest HEAD could not hold in memory anymore a Mistral 7B training on tpu-v3-8, sequence length was set to 2048
Git bisect indicated that the commit below was the first with RESOURCE_EXHAUSTED: XLA:TPU

```commit 28d53a4e98a2f1c5ae5d53581b5cbe761be004a8 (refs/bisect/bad)
Author: erfanzar <erfanzare82@yahoo.com>
Date:   Wed Jan 31 12:02:38 2024 -0800

    Working on `LoRA`
```
The preceding commit does not exhibit the resource exhausted issue.

```commit b296163de12cfadb243d61e7fed9c8ecde0b45fd (refs/bisect/good-b296163de12cfadb243d61e7fed9c8ecde0b45fd)
Author: Erfan Zare Chavoshi <59269023+erfanzar@users.noreply.github.com>
Date:   Wed Jan 31 18:33:47 2024 +0330

    Delete python_test/EasyDel-Checkpoints/Lora-Test directory
```

I instantiate the trainarguments like this (I've incorporated the change from max_length to max_sequence_length) :
```
tokenizer = AutoTokenizer.from_pretrained(model_id, trust_remote_code=True)
model, params = AutoEasyDelModelForCausalLM.from_pretrained(model_id)
config = model.config
config.freq_max_position_embeddings = config.max_position_embeddings  # 32768
config.max_position_embeddings = 2048  # Let use context length of ... for training
config.c_max_position_embeddings = config.max_position_embeddings

import pprint
pprint.pprint(config.get_partition_rules(True))

max_sequence_length = config.max_position_embeddings

train_args = TrainArguments(
    model_class=EasyDel.FlaxMistralForCausalLM,
    configs_to_initialize_model_class={
        'config': config,
        'dtype': jnp.bfloat16,
        'param_dtype': jnp.bfloat16,
        'input_shape': (1, 1)
    },
    custom_rule=config.get_partition_rules(True),
    model_name='MistralBoreas',
    num_train_epochs=1,
    learning_rate=1e-5,
    learning_rate_end=1e-6,
    optimizer=EasyDelOptimizers.ADAMW,
    scheduler=EasyDelSchedulers.WARM_UP_LINEAR,
    warmup_steps=500,
    weight_decay=0.01,
    total_batch_size=1,
    save_steps=2000,
    do_train=True,
    do_eval=False,
    backend='tpu',
    max_sequence_length=max_sequence_length,
    gradient_checkpointing=EasyDelGradientCheckPointers.NOTHING_SAVEABLE,
    sharding_array=(1, -1, 1, 1),
    use_pjit_attention_force=False,
    use_flash_attention=False,
    gradient_accumulation_steps=8,
    remove_ckpt_after_load=True,
    ids_to_pop_from_dataset=['token_type_ids'],
    loss_remat="",
    dtype=jnp.bfloat16
)

trainer = CausalLanguageModelTrainer(
    train_args,
    dataset_train,
    checkpoint_path=None
)
```

**Comment by erfanzar** [comment]
Hello 
Thank for reporting the issue ill fix that asap.

**Comment by erfanzar** [comment]
fixed you can try training `Mistral` Model on kaggle with a batch size of 8 and gradient au of 4.

and thank for reporting the issue

**Comment by yhavinga** [comment]
Verified it works, thanks!
